code/ship/shiphit.cpp

```diff
@@ -1882,6 +1885,39 @@ static int maybe_shockwave_damage_adjust(object *ship_objp, object *other_obj, f
 		max_damage *= 4.0f;
 	}
 
+	// If the shockwave was caused by a weapon, then check if the weapon can deal lethal damage to the ship.
+	// If it cannot, then neither should the shockwave caused by the weapon be able to do the same.
+	// The code for this is copied from part of weapon_get_damage_scale.
+	int wp_index = shockwave_get_weapon_index(other_obj->instance);
+	if ((wp_index >= 0) && Weapon_shockwaves_respect_huge) {
+		wip = &Weapon_info[wp_index];
+
+		float hull_pct = get_hull_pct(ship_objp);
+
+		// First handle Supercap ships.
+		if ((sip->flags[Ship::Info_Flags::Supercap]) && !(wip->wi_flags[Weapon::Info_Flags::Supercap])) {
+			if (hull_pct <= 0.75f) {
+				*damage = 0.0f;
+				return 1;
+			} else {
+				// If hull isn't below 3/4, then allow damage to be applied just like in weapon_get_damage_scale.
+				// Magic constant is SUPERCAP_DAMAGE_SCALE from weapons.cpp.
+				max_damage *= 0.25f;
```

Comment:
This should be the aforementioned constant rather than a magic number with the same value.

Reply:
Where should I define said constant? It's defined in weapons.cpp and not in the header so I had to copy it. Should it be in shiphit.cpp or in shiphit.h?

Reply:
obviously, you move it to the header so it can be reused

Reply:
Should I add a prefix to it since the macro will then be global? Like, should I change it to WEAPON_H_SUPERCAP_DAMAGE_SCALE  ?

Reply:
It shouldn't cause any ambiguity, so no.